server/src/main/java/org/opensearch/indices/store/TransportNodesListShardStoreMetadata.java

```diff
@@ -176,7 +176,7 @@ public Request(StreamInput in) throws IOException {
         }
 
         public Request(ShardId shardId, String customDataPath, DiscoveryNode[] nodes) {
-            super(nodes);
+            super(false, nodes);
```

Comment:
this change to send discoveryNodes was introduced for replica shard store fetch, we may have to keep it in this flow?